api/src/main/java/jakarta/enterprise/lang/model/types/ParameterizedType.java

```diff
@@ -4,20 +4,20 @@
 import java.util.List;
 
 /**
- * A parameterized type. That is, a {@linkplain #declaration() generic class} together with
+ * A parameterized type. That is, a {@linkplain #genericClass() generic class} together with
  * a list of {@linkplain #typeArguments() type arguments}. The list of type arguments
  * has the same shape as the class's list of type parameters.
  *
  * @since 4.0
  */
 public interface ParameterizedType extends Type {
     /**
-     * Returns the {@linkplain ClassInfo declaration} of the generic class that was parameterized with a list
+     * Returns the {@linkplain ClassType type} of the generic class that was parameterized with a list
      * of type arguments to form this parameterized type.
      *
-     * @return the {@linkplain ClassInfo declaration} of the generic class, never {@code null}
+     * @return the {@linkplain ClassType type} of the generic class, never {@code null}
      */
-    ClassInfo declaration();
+    ClassType genericClass();
```

Comment:
This comes from me thinking how to describe what a type is. Parameterized type isnt' created by applying a list of type arguments to a class, but to a class _type_. So I thought that should be a first-class concept here. Allowing access to the declaration still makes sense, so that is retained.

Reply:
I realized that I got this one wrong. The type arguments are not applied to the class _type_; it's applied to the _type constructor_ that is introduced by the class declaration. I need to think more about this :-)